src/systems/triggered_publisher/TriggeredPublisher.cc

```diff
@@ -645,6 +703,71 @@ void TriggeredPublisher::Configure(const Entity &,
       std::thread(std::bind(&TriggeredPublisher::DoWork, this));
 }
 
+//////////////////////////////////////////////////
+void TriggeredPublisher::PublishMsg(std::size_t pending)
+{
+    for (auto &info : this->outputInfo)
+    {
+      for (std::size_t i = 0; i < pending; ++i)
+      {
+        info.pub.Publish(*info.msgData);
+      }
+    }
+}
+
+//////////////////////////////////////////////////
+void TriggeredPublisher::CallService()
+{
+  for (auto &serviceInfo : this->servOutputInfo)
+  {
+    bool result;
+    auto req = msgs::Factory::New(serviceInfo.reqType,
+                                  serviceInfo.reqMsg);
+    if (!req)
+    {
+      ignerr << "Unable to create request for type ["
+             << serviceInfo.reqType << "].\n";
+      return;
+    }
+
+    auto rep = msgs::Factory::New(serviceInfo.repType);
+    if (!rep)
+    {
+      ignerr << "Unable to create response for type ["
+             << serviceInfo.repType << "].\n";
+      return;
+    }
+
+    bool executed = this->node.Request(serviceInfo.servName,
+                                       *req,
+                                       serviceInfo.timeout,
+                                       *rep,
+                                       result);
+    if (executed)
+    {
+      if (!result)
+      {
+        ignerr << "Service call [" << serviceInfo.servName
+               << "] failed\n";
```

Comment:
A lot of these messages are under 80 chars in one line. We don't need to take up extra lines.